### PR TITLE
#155931587 Accessing the API

### PR DESCRIPTION
**Pull request by bibangamba**
#### What does this PR do?
Add implementation to access the GitHub API

#### Description of tasks to be done?
- add a GithubUsersView interface that will be used to connect the presenter to the MainActivity
- add a GithubUsersPresenter class that will make use of the retrofit instance to request data from the API and convert it into a list of GithubUser objects

#### How can this be manually tested?
check for the presence of a GithubUsersView interface and a GithubUsersPresenter class

#### What are the relevant pivotal tracker stories?
[#155931587](https://www.pivotaltracker.com/story/show/155931587)

### CODE REVIEW

File: app/src/main/java/presenter/GithubUsersPresenter.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+    public void getGithubUsers() {
+        githubService
+                .getApi()
+                .getAllUsers()
+                .enqueue(new Callback<GithubUsersResponse>() {
+                    @Override
+                    public void onResponse(@NonNull Call<GithubUsersResponse> call, @NonNull Response<GithubUsersResponse> response) {
+                        GithubUsersResponse githubUsersResponse = response.body();
+                        if (githubUsersResponse != null && githubUsersResponse.getUsers() != null) {
```
Comment (joshNic) [inline app/src/main/java/presenter/GithubUsersPresenter.java]: Please refrain from using != operator when doing logical operations

File: app/src/main/java/view/GithubUsersView.java [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package view;
+
+import java.util.List;
+
+import model.GithubUsers;
+
+public interface GithubUsersView {
+    void githubUsersReady(List<GithubUsers> githubUsers);
```
Comment (joshNic) [inline app/src/main/java/view/GithubUsersView.java]: Please try to follow the [clean code conventions](https://andela.instructure.com/courses/282/files/22422/download?verifier=JdA53jXl4Q9FgM4zWuevMBB1Dli95Pt0yyOBnu7S&wrap=1) 
when naming methods and variables

File: app/src/main/java/presenter/GithubUsersPresenter.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+import model.GithubUsers;
+import model.GithubUsersResponse;
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+import service.GithubService;
+import view.GithubUsersView;
+
+public class GithubUsersPresenter {
+    private static final String TAG = "GithubUsersPresenter";
```
Comment (joshNic) [inline app/src/main/java/presenter/GithubUsersPresenter.java]: Could you please put this string variable in the String resource file such that you just reference it here

File: app/src/main/java/presenter/GithubUsersPresenter.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    private GithubUsersView githubUsersView;
+    private GithubService githubService;
+
+    public GithubUsersPresenter(GithubUsersView view) {
+        this.githubUsersView = view;
+        if (this.githubService == null) {
```
Comment (joshNic) [inline app/src/main/java/presenter/GithubUsersPresenter.java]: Please refrain from using the == operator you can use the .equals() java method

**Review by joshNic — CHANGES_REQUESTED** [review]
Hello @bibangamba this is good work I like that in some parts of your code you have tried to conform to the clean code conventions but there are some code smells which I have pointed out, please attend to those ASAP so that I can clear this. Thank you.

**Comment by bibangamba** [comment]
Thanks for the feedback (and the clean code resource), I have implemented it

**Review by joshNic — APPROVED** [review]
LGTM

**Pull request merged**